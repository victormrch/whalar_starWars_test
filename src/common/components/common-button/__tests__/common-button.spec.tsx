import { CommonButton } from '../index';
import { render, screen } from '@testing-library/react';

describe('CommonButton component specs', () => {
	it('should render correctly and create snapshot', () => {
		const message = 'Test';

		const { asFragment } = render(<CommonButton buttonName={message} />);

		expect(asFragment()).toMatchSnapshot();
	});
	it('should display the message when pass the buttonName prop', () => {
		const message = 'Test Message';

		render(<CommonButton buttonName={message} />);

		const element = screen.getByRole('button', {
			name: /test message/i,
		});

		expect(element).not.toBeNull();
		expect(element).toBeInTheDocument();
		expect(element.tagName).toEqual('BUTTON');
		expect(element.textContent).toBe('Test Message');
	});
});
