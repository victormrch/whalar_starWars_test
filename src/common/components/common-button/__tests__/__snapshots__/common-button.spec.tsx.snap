// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonButton component specs should render correctly and create snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ kjcozG"
  >
    <button
      class="sc-gsnTZi fDlIUR"
    >
      Test
    </button>
  </div>
</DocumentFragment>
`;
